### PR TITLE
DM-worker: fix toml decode time.duration in config

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

`time.Duration` is type alias for int64, the toml library we use doesn't support decode `time.Duration` directly

### What is changed and how it works?

Create a new `duration` struct which embeds the `time.Duration`
- Hack the `UnmarshalText` method for toml decoding
- Hack the `MarshalJSON` for json format string

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note
   - Optimize time.Duration type parameter in the configuration of DM-worker

### CODE REVIEW

File: dm/worker/task_checker.go [hunk 1]
```diff
@@ -90,21 +91,47 @@ func (bs ResumeStrategy) String() string {
...
+func (d duration) UnmarshalText(text []byte) error {
+	var err error
+	d.Duration, err = time.ParseDuration(string(text))
+	return err
+}
+
+// MarshalJSON hacks to satisfy the json.Marshaler interface
+func (d *duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Duration string `json:"duration"`
```
Comment (csuzhangxc) [inline dm/worker/task_checker.go]: we have JSON flag in L96, is this still needed?

Comment (amyangfei) [reply dm/worker/task_checker.go]: yes, the JSON tag here is must, which is really used during JSON marshal

Comment (csuzhangxc) [reply dm/worker/task_checker.go]: can we remove this `MarshalJSON` method and only rely on the JSON flag in L96?

Comment (amyangfei) [reply dm/worker/task_checker.go]: If we don't implement this method, the JSON output will be `int64` value of `Duratioin`  in nanosecond level, which is not human readable

**Comment by csuzhangxc** [comment]
@WangXiangUSTC PTAL

File: dm/worker/task_checker.go [hunk 2]
```diff
@@ -90,21 +91,47 @@ func (bs ResumeStrategy) String() string {
 	return fmt.Sprintf("unsupported resume strategy: %d", bs)
 }
 
+// duration is used to hold a time.Duration field
+type duration struct {
+	time.Duration `toml:"duration" json:"duration"`
```
Comment (WangXiangUSTC) [inline dm/worker/task_checker.go]: why need `toml:"duration"`?

Comment (amyangfei) [reply dm/worker/task_checker.go]: removed

**Comment by amyangfei** [comment]
PTAL @WangXiangUSTC

**Pull request merged**